step/step.go

```diff
@@ -0,0 +1,113 @@
+package step
+
+import (
+	"fmt"
+
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/command"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-steplib/steps-git-clone/gitclone"
+)
+
+type Input struct {
+	ShouldMergePR bool `env:"merge_pr,opt[yes,no]"`
+
+	CloneIntoDir         string   `env:"clone_into_dir,required"`
+	CloneDepth           int      `env:"clone_depth"`
+	UpdateSubmodules     bool     `env:"update_submodules,opt[yes,no]"`
+	SubmoduleUpdateDepth int      `env:"submodule_update_depth"`
+	FetchTags            bool     `env:"fetch_tags,opt[yes,no]"`
+	SparseDirectories    []string `env:"sparse_directories,multiline"`
+
+	RepositoryURL         string `env:"repository_url,required"`
+	Commit                string `env:"commit"`
+	Tag                   string `env:"tag"`
+	Branch                string `env:"branch"`
+	PRDestBranch          string `env:"branch_dest"`
+	PRSourceRepositoryURL string `env:"pull_request_repository_url"`
+	PRMergeBranch         string `env:"pull_request_merge_branch"`
+	PRHeadBranch          string `env:"pull_request_head_branch"`
+
+	ResetRepository bool   `env:"reset_repository,opt[Yes,No]"`
+	BuildURL        string `env:"build_url"`
+	BuildAPIToken   string `env:"build_api_token"`
+}
+
+// Config is the git clone step configuration
+type Config struct {
+	Input
```

Comment:
Why do we have a `Config` struct here if there is another struct in the `gitclone` package?

Reply:
Hmm, yeah, it's redundant, but we wanted to conform to the ideal architecture while treating most of the existing code as a library for the "step" to call. This is to minimize the changes we make in this PR due to this step being so critical for all of our users.

Therefore we left the other struct in the `gitclone` package and just mapped this config to that one since this config is the one that gets parsed.